### PR TITLE
Fix a deserialization problem in SszBitvector

### PR DESCRIPTION
**Pull request by jtraglia**
## PR Description

I was creating property tests for all of the serialization/deseriation classes and came across this issue. When the serialized value is negative, the bit length is incorrect. I will share the property tests later after I fix some more potential bugs.

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by jtraglia** [comment]
Hold off on this. Something is not quite right.

File: build.gradle [hunk 1]
```diff
@@ -176,6 +176,7 @@ allprojects {
       '-Xlint:finally',
       '-Xlint:static',
       '-Werror',
+      '-parameters',
```
Comment (jtraglia) [inline build.gradle:179]: Added this so that parameter names show up in property test failure output.

**Comment by jtraglia** [comment]
So my original "fix" misidentified the problem and I don't know what the proper fix should be. I've added the tests I used to identify the problem. These are expected to fail. I'll continue to look at it, but if I can't get it I might need some help.

**Comment by jtraglia** [comment]
I couldn't figure out how to correctly compute the bit length given the original serialized value, so I decided to redo it in a way that made sense to me. It is now serialized to a string of 0 and 1 values, like its `toString` output. The serialized value is less space efficient but potentially easier to read if required. Not sure what this will break but it's probably a lot. If someone could fix the previous implementation that would be preferred.

I think the problem is because the bitvector size is lost in translation. In other usages, the bit length is already known when deserializing. For example, the following:

https://github.com/ConsenSys/teku/blob/516444f5b43ff85530d32e7a3206acc369a8b8fb/infrastructure/ssz/src/test/java/tech/pegasys/teku/infrastructure/ssz/collections/SszBitvectorTest.java#L73-L74

It's my understanding that `.getSchema()` will keep the bitvector size.

**Comment by ajsutton** [comment]
I'll need to look into the SSZ spec better to work this out.  It's definitely a bit of an issue trying to parse a Bitvector without actually having the right schema so it may not be possible to make work right in every case.  On the plus side we're migrating away from that form of JSON parsing and I think we may have already migrated the only APIs that use Bitvector. If not we could prioritise them and delete this implementation I suspect.

Will look in more detail Monday - it's going to take more focus than I'll get over the weekend. :)

**Comment by jtraglia** [comment]
Sounds good! We'll deal with it then. Have a good weekend.

**Comment by ajsutton** [comment]
Have put up #5555 to remove the deserialiser entirely since it isn't used in production anymore. I'm not sure there is a way to make it work reliably given it doesn't have the actual vector length from the schema.

**Comment by jtraglia** [comment]
With that PR this is no longer an issue. Thanks!

**Comment by jtraglia** [comment]
I will commit these new property tests in a separate PR.

**Pull request closed without merging**